packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -229,7 +231,11 @@ export class HbarLimitService implements IHbarLimitService {
       }
     }
     if (ipAddress) {
-      // TODO: Implement this with https://github.com/hashgraph/hedera-json-rpc-relay/issues/2888
+      try {
+        return await this.getSpendingPlanByIPAddress(ipAddress);
+      } catch (error) {
+        this.logger.warn(error, `Failed to get spending plan for IP address '${ipAddress}'`);
```

Comment:
shouldnt this be this.logger.error, since we have an error? also i think we should throw an error and whatever method calls this can catch it and propagate it

Reply:
It's a warning because not finding an existing plan at this point is not critical and a new one is created later in the workflow.